src/lib/decorations/path.typ

```diff
@@ -59,6 +63,16 @@
   factor: 150%,
 )
 
+#let resolve-amplitude(amplitude, segment, num-segments) = {
+  return if type(amplitude) == function {
+    (amplitude)(segment / num-segments * 100%)
```

Comment:
I think it's better if this is `segment / (num-segments - 1) * 100%`, as then the argument will range from `0%` to `100%` inclusive for the first and last segments. Then the output will appear symmetric for symmetric amplitude functions like `t => calc.min(float(t), 1 - float(t))`.